next/TilesetWithFullMetadata/TilesetWithFullMetadata.json

```diff
@@ -0,0 +1,1318 @@
+                "type" : "ARRAY",
+                "componentType" : "BOOLEAN",
+                "componentCount" : 5
+              },
+              "example_variable_length_ARRAY_STRING" : {
+                "name" : "Example variable-length ARRAY STRING property",
+                "description" : "An example property, with type ARRAY, with component type STRING, and variable length ",
+                "type" : "ARRAY",
+                "componentType" : "STRING"
+              },
+              "example_fixed_length_ARRAY_STRING" : {
+                "name" : "Example fixed-length ARRAY STRING property",
+                "description" : "An example property, with type ARRAY, with component type STRING, and fixed length ",
+                "type" : "ARRAY",
+                "componentType" : "STRING",
+                "componentCount" : 5
+              },
+              "example_variable_length_ARRAY_ENUM" : {
+                "name" : "Example variable-length ARRAY ENUM property",
+                "description" : "An example property, with type ARRAY, with component type ENUM, enum type exampleEnumType, and variable length ",
+                "type" : "ARRAY",
+                "enumType" : "exampleEnumType",
+                "componentType" : "ENUM"
+              },
+              "example_fixed_length_ARRAY_ENUM" : {
+                "name" : "Example fixed-length ARRAY ENUM property",
+                "description" : "An example property, with type ARRAY, with component type ENUM, enum type exampleEnumType, and fixed length ",
+                "type" : "ARRAY",
+                "enumType" : "exampleEnumType",
+                "componentType" : "ENUM",
+                "componentCount" : 5
+              }
+            }
+          }
+        },
+        "enums" : {
+          "exampleEnumType" : {
+            "values" : [ {
+              "name" : "ExampleEnumValueA",
+              "value" : 0
+            }, {
+              "name" : "ExampleEnumValueB",
+              "value" : 1
+            }, {
+              "name" : "ExampleEnumValueC",
+              "value" : 2
+            } ]
+          }
+        }
+      },
+      "tileset" : {
+        "class" : "exampleClass",
+        "properties" : {
+          "example_INT8" : -128,
+          "example_normalized_INT8" : -128,
+          "example_UINT8" : 255,
+          "example_normalized_UINT8" : 255,
+          "example_INT16" : -32768,
+          "example_normalized_INT16" : -32768,
+          "example_UINT16" : 65535,
+          "example_normalized_UINT16" : 65535,
+          "example_INT32" : -2147483648,
+          "example_normalized_INT32" : -2147483648,
+          "example_UINT32" : 4294967295,
+          "example_normalized_UINT32" : 4294967295,
+          "example_INT64" : -9223372036854775808,
+          "example_normalized_INT64" : -9223372036854775808,
+          "example_UINT64" : 18446744073709551615,
```

Comment:
yikes, I see now that UINT64 doesn't work well here due to the fact that JSON parsing doesn't use `BigInt`

CesiumJS does use BigInt for binary properties, but from JSON there's no way to do this since values are encoded as `number`s...

this might be something to bring up for discussion, does it make sense to have 64-bit types in JSON? Though I suppose one could argue that other languages parse 64-bit integers just fine, it's really a JS limitation here.

Reply:
This is the point where one can dive into some philsophical discussion (as in: "Samples" are intended to show things that work. "Tests" are intended to show things that do _not_ work). 

The difficulties that may arise from integer values outside the 2<sup>53</sup> range had also been an issue in glTF (see https://github.com/KhronosGroup/glTF/pull/1574#issuecomment-570195661 etc). Right now, the spec does not say much _explicitly_ about that, but points to the [JSON spec,](https://www.rfc-editor.org/rfc/rfc8259.txt) which says, for example,
 
> Note that when such software is used, numbers that are integers and are in the range [-(2**53)+1, (2**53)-1] are interoperable in the sense that implementations will agree exactly on their numeric values.

Eventually, one cannot assume that software will be able to correctly handle these numbers. In how far we address this is also the question that I mentioned in the opening comment of this PR.